scriptmodules/supplementary/emulationstation/configscripts/mupen64plus.sh

```diff
@@ -0,0 +1,267 @@
+                        value="${ini_value}${input_id})"
+                    elif [[ "$ini_value" == *\)* ]]; then
+                        value="button(${input_id},${ini_value}"
+                    elif [[ "$dir" == "Up" || "$dir" == "Left" ]]; then
+                        value="button(${input_id},"
+                    elif [[ "$dir" == "Right" || "$dir" == "Down" ]]; then
+                        value="${input_id})"
+                    fi
+                else
+                    value="button(${input_id}) ${ini_value}"
+                fi
+                ;;
+        esac
+
+        iniSet "$key" "$value"
+    done
+}
+
+function onend_mupen64plus_joystick() {
+    local device_type=$1
+    local device_name=$2
+
+    local bind
+    local axis
+    local axis_neg
+    local axis_pos
+    for axis in "X Axis" "Y Axis"; do
+        if [[ "$axis" == *X* ]]; then
+            axis_neg="DPad L"
+            axis_pos="DPad R"
+        else
+            axis_neg="DPad U"
+            axis_pos="DPad D"
+        fi
+
+        # analog stick sanity check
+        # replace Axis values with DPAD values if there is no Axis 
+        # device setup
+        if ! grep -q "$axis" /tmp/mp64tempconfig.cfg ; then
+          iniGet "${axis_neg}"
+          bind=${ini_value//)/,}
+          iniGet "${axis_pos}"
+          ini_value=${ini_value//axis(/}
+          ini_value=${ini_value//hat(/}
+          ini_value=${ini_value//button(/}
+          bind="${bind}${ini_value}"
+          iniSet "$axis" "$bind"
+          iniDel "${axis_neg}"
+          iniDel "${axis_pos}"
+        fi
+    done
+
+    echo "; ${device_name}_END " >> /tmp/mp64tempconfig.cfg
+    echo "" >> /tmp/mp64tempconfig.cfg
+
+    # abort if old device config cannot be deleted.
+    # keep original mupen64plus-input-sdl configs.
+    local file="/opt/retropie/configs/n64/InputAutoCfg.ini"
+    if [[ -f "$file" ]]; then
+        # backup current config file
+        cp "$file" "${file}.bak"
+        sed -i /"${device_name}_START"/,/"${device_name}_END"/d "$file"
+        if grep -q "$device_name" "$file" ; then
+            rm /tmp/mp64tempconfig.cfg
+            return
+        fi
+    else
+        cat > "$file" << _EOF_
+; InputAutoCfg.ini for Mupen64Plus SDL Input plugin
+
+; Keyboard_START
+[Keyboard]
+plugged = True
+plugin = 2
+mouse = False
+DPad R = key(100)
+DPad L = key(97)
+DPad D = key(115)
+DPad U = key(119)
+Start = key(13)
+Z Trig = key(122)
+B Button = key(306)
+A Button = key(304)
+C Button R = key(108)
+C Button L = key(106)
+C Button D = key(107)
+C Button U = key(105)
+R Trig = key(99)
+L Trig = key(120)
+Mempak switch = key(44)
+Rumblepak switch = key(46)
+X Axis = key(276,275)
+Y Axis = key(273,274)
+; Keyboard_END
+
+_EOF_
+    fi
+
+    # read temp device configuration and append to InputAutoCfg.ini
+    local deviceconfig=`cat /tmp/mp64tempconfig.cfg`
+    echo "$deviceconfig" >> "$file"
+    rm /tmp/mp64tempconfig.cfg
```

Comment:
I am about to change this block of code, but wanted to check just in case there is something I have overlooked - Why is there a variable used here over just doing `cat /tmp/mp64tempconfig.cfg >>"$file"` - going via a variable and using echo will strip the LF I believe - was that needed ?